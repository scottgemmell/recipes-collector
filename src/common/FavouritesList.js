import React from "react";
import PropTypes from "prop-types";
import RecipesThumb from "./RecipesThumb";

const FavouritesList = ({ handleFavourite, list, myFavourites }) => {
	return (
		<ul className="list">
			{list.map(item => (<RecipesThumb 
				key={item.idMeal} 
				{...item} 
				isFav={myFavourites.includes(+item.idMeal)}
				handleFavourite={handleFavourite} 
			/>)
			)}
		</ul>
	);
};

FavouritesList.propTypes = {
	handleFavourite: PropTypes.func.isRequired, 
	list: PropTypes.array.isRequired, 
	myFavourites: PropTypes.array,
};

export default FavouritesList;
