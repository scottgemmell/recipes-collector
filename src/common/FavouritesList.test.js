import React from "react";
import { shallow } from "enzyme";
import FavouritesList from "./FavouritesList";

const defaultProps = () => ({});
const setup = (overrideProps = {}) => {
	const props = Object.assign({}, defaultProps(), overrideProps);
	const wrapper = shallow(<FavouritesList {...props} />);
	return { wrapper, props };
};

describe("<FavouritesList />", () => {
	const handleFavourite = jest.fn();
	it("renders", () => {
		const { wrapper } = setup({ 
			handleFavourite, 
			list: [{ 
				idMeal: "100",
				strMeal: "something",
				strMealThumb: "something",
				isLoading: false,
			}], 
			myFavourites: ["100"],
		});
		expect(wrapper).toMatchSnapshot();
	});
});
