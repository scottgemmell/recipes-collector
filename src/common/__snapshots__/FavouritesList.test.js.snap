// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FavouritesList /> renders 1`] = `ShallowWrapper {}`;
